base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1405 @@
+!!      ! ygyu test NetCDF4_get_var.H for group
+!!      test_get_var_group: block
+!!        type(NetCDF4_FileFormatter) :: formatter
+!!        type(FileMetadata) :: fmd_new
+!!        real(kind=REAL32), allocatable :: data_real32(:)
+!!        integer :: ncid, grpid, ncid0
+!!        integer :: dimid(10),  dimlen(10)        
+!!        integer :: len
+!!        character(len=60) :: grp_name
+!!        character(len=60) :: var_name, dim_name(10)
+!!        
+!!        call formatter%open(filename, pFIO_READ, _RC)
+!!        fmd_new=formatter%read(_RC)
+!!        grp_name='MetaData'
+!!        var_name='latitude'
+!!        dim_name(1)='Channel'
+!!        dim_name(2)='Location'
+!!        ncid0=formatter%ncid
+!!        call check_nc_status( nf90_inq_ncid(ncid0, grp_name, ncid), _RC)
+!!        grpid=ncid
+!!        do i=1, 2
+!!           call check_nc_status( nf90_inq_dimid(ncid, dim_name(i), dimid(i)), _RC)
+!!           call check_nc_status( nf90_inquire_dimension(ncid, dimid(i), len=dimlen(i)), _RC)
+!!        end do
+!!
+!!        len=dimlen(2)
+!!        allocate (data_real32(len))
+!!        call formatter%get_var(var_name, data_real32, group_name=grp_name, count=[len], rc=status)
+!!        write(6,*) 'af get_var:  data_real32'
+!!        write(6,*) data_real32
+!!        stop -11
+!!      end block test_get_var_group
+!!-----------------------------------------------------------
+
+      
+      
+      ! read global dim from the whole nc file
+      ! --------------------------------------
+      key_lon='cell_across_swath'
+      key_lat='cell_along_swath'
+      key_time='time'
+      CALL get_ncfile_dimension(filename, nlon, nlat, tdim, key_lon, key_lat, key_time, _RC)
+      allocate(scanTime(nlon, nlat))
+      allocate(this%t_alongtrack(nlat))
+      write(6,*) 'filename', trim(filename)
+      write(6,*) 'nlon,nlat,tdim', nlon,nlat,tdim      
+      call get_v2d_netcdf(filename, 'scanTime', scanTime, nlon, nlat)
+      do j=1, nlat
+         !!i=j*nlon+1
+         this%t_alongtrack(j)= scanTime(1,j)
+         if ( mod(j,100)== 1) write(6,*)  j, scanTime(1,j)
+      enddo
+      !
+      ! skip un-defined time value
+      !
+      !
+      nstart = 1
+      x0 = this%t_alongtrack(1)
+      x1 = 1.d16
+
+      if (x0 > x1) then
+         !
+         ! bisect backward finding the first index arr[n] < x1
+         klo=1
+         khi=nlat
+         max_iter = int( log( real(nlat) ) / log(2.d0) ) + 2         
+         do i=1, max_iter
+            k = (klo+khi)/2
+            if ( this%t_alongtrack(k) < x1 ) then
+               khi=k
+            else
+               nstart = khi
+               exit
+            endif
+         enddo         
+         write(6,*) 'nstart', nstart
+         write(6,*) ' this%t_alongtrack(nstart)',  this%t_alongtrack(nstart)
+      endif
+      
+      deallocate(scanTime)
+      write(6,*) 't_a'
+      write(6,*) this%t_alongtrack(nstart:nstart+10)
```

Comment:
Again - any text IO that we want to keep should be converted to pFlogger

Reply:
I have converted these cases to pFlogger. Thanks.